engine/execution/messages.go

```diff
@@ -53,60 +133,104 @@ func NewEmptyComputationResult(
 	}
 }
 
-func (cr ComputationResult) transactionResultsByCollectionIndex(colIndex int) []flow.TransactionResult {
-	var startTxnIndex int
-	if colIndex > 0 {
-		startTxnIndex = cr.TransactionResultIndex[colIndex-1]
-	}
-	endTxnIndex := cr.TransactionResultIndex[colIndex]
-	return cr.TransactionResults[startTxnIndex:endTxnIndex]
-}
-
-func (cr *ComputationResult) CollectionResult(colIndex int) *ColResSnapshot {
-	if colIndex < 0 && colIndex > len(cr.CompleteCollections) {
+func (cr *BlockAttestationResults) AttestedCollection(colIndex int) *CollectionAttestationResult {
+	if colIndex < 0 && colIndex > len(cr.Chunks) {
 		return nil
```

Comment:
Better to return `nil, error`

Reply:
Yeah, I initially started with that but that needs a lot of changes right now in the code base, I capture it as a TODO for the future when we actually split the attestation from execution.